src/restruct_poc/model/node.hpp

```diff
@@ -17,6 +18,28 @@ struct Node {
         Array_6 acceleration = Array_6{0., 0., 0., 0., 0., 0.}
     )
         : ID(id), x(position), u(displacement), v(velocity), vd(acceleration) {}
+
+    /// Translate a node by a displacement vector
+    void Translate(const Array_3& displacement) {
+        x[0] += displacement[0];
+        x[1] += displacement[1];
+        x[2] += displacement[2];
+    }
+
+    /// Rotate a node by a rotation axis and angle
+    void Rotate(const Array_3& axis, double angle) {
+        auto q = Array_4{
+            cos(angle / 2.), sin(angle / 2.) * axis[0], sin(angle / 2.) * axis[1],
+            sin(angle / 2.) * axis[2]};
+        auto R = QuaternionToRotationMatrix(q);
```

Comment:
Could we use `RotateVectorByQuaternion` instead of converting to a rotation matrix? Also, I'd like to specify the point about which the node is being rotated. I think this should affect the node's rotation component as well, so you'll need to compose the node's current rotation with the quaternion that's constructed in this function. I also think this needs to modify the velocity and acceleration vectors.

Here's the test case. Create a blade in it's default orientation along the X axis with the blade root at the hub radius of 1 meters. Rotate about the blade root to give it a cone angle of 5 degrees (rotate around Y axis).  Set the rotational velocity to 5 rad/s about the Z axis (this may take some figuring out). Rotate +90 degrees around Z axis so that the blade is pointing up along +Y. Tilt the blade back to account for 5 degrees of shaft tilt (-5 degrees about X axis). Translate along +Y by 100 meters. The question is, what is the tip velocity. We may need to discuss and calculate this value. Also there may be a better order of operations, I'm open to other ideas.

Reply:
> Could we use RotateVectorByQuaternion instead of converting to a rotation matrix? Also, I'd like to specify the point about which the node is being rotated. I think this should affect the node's rotation component as well, so you'll need to compose the node's current rotation with the quaternion that's constructed in this function. I also think this needs to modify the velocity and acceleration vectors.

I refactored the `Rotate` method to accomplish this, however not sure what you mean by specifying the point of rotation -- it needs to be a vector and and angle right?